### PR TITLE
[Confluence] Documentation for Confluence connector

### PR DESCRIPTION
**Pull request by praveen-elastic**
## Checklists

Parent issue https://github.com/elastic/connectors-python/issues/541
#### Pre-Review Checklist
- [x] Considered corresponding documentation changes

### CODE REVIEW

File: docs/DEVELOPING.md
Comment (danajuratoni) [inline docs/DEVELOPING.md:777]: We should not use defaults where they're not helping users e.g. username, password etc.
I created https://github.com/elastic/connectors-python/issues/606 for adjusting existing connectors, let's make sure new connectors will align to those rules, and we document examples where necessary

File: docs/DEVELOPING.md
Comment (praveen-elastic) [reply docs/DEVELOPING.md:827]: @danajuratoni the change requires using new set of API endpoints for the connector since currently pages, blogposts and attachments are fetched independent of the space.

Since this is a scope change, can you please create an enhancement issue and we'll take it once the present one is merged on main.

File: docs/DEVELOPING.md
Comment (praveen-elastic) [reply docs/DEVELOPING.md:777]: As of now default values are used to run e2e and the PR for similar change in Jira and SharePoint is still draft https://github.com/elastic/connectors-python/pull/648, shall we wait for this to get merged ?

File: docs/DEVELOPING.md
Comment (praveen-elastic) [reply docs/DEVELOPING.md:793]: In MySQL and Generic DB we were using `host` and `port` as two different fields and making a connection string out of those fields but in the case of Jira and Confluence port is the part of the host that's why we have consider it as a whole `host_url`
WDYT?

File: docs/DEVELOPING.md
Comment (danajuratoni) [inline docs/DEVELOPING.md:785]: This seems to be the email address for the Atlassian account users will be using to create the token below. `_id` does not seem to fit for this field

File: docs/DEVELOPING.md
Comment (danajuratoni) [reply docs/DEVELOPING.md:827]: Sure, this can be added as an enhancement. Created https://github.com/elastic/connectors-python/issues/659

File: docs/DEVELOPING.md
Comment (praveen-elastic) [reply docs/DEVELOPING.md]: [6d316f7](https://github.com/elastic/connectors-python/pull/647/commits/6d316f71c195ad4a7d3ccb23819882b9c9e462e3)

File: docs/DEVELOPING.md
Comment (praveen-elastic) [reply docs/DEVELOPING.md:59]: [6d316f7](https://github.com/elastic/connectors-python/pull/647/commits/6d316f71c195ad4a7d3ccb23819882b9c9e462e3)

File: docs/DEVELOPING.md
Comment (praveen-elastic) [reply docs/DEVELOPING.md]: Updated with commit id: [6d316f7](https://github.com/elastic/connectors-python/pull/647/commits/6d316f71c195ad4a7d3ccb23819882b9c9e462e3)

**Comment by danajuratoni** [comment]
LGTM
@navarone-feekery once work for Rich configurable fields is done, it would be helpful to have the labels documented as well, since labels are the ones showing up in Kibana.

cc: @leemthompo @ppf2 - Confluence doc made good progress and it follows the official format. Feel free to take a look!

File: docs/DEVELOPING.md
Comment (leemthompo) [inline docs/DEVELOPING.md:52]:
```suggestion
```

File: docs/DEVELOPING.md
Comment (leemthompo) [inline docs/DEVELOPING.md:57]:
```suggestion
Dropdown to determine the Confluence platform type: `Confluence Cloud` or `Confluence Server`. Default value is `Confluence Cloud`.
```

**Review by leemthompo — APPROVED** [review]
LGTM with a few minor changes

File: docs/DEVELOPING.md
Comment (ppf2) [inline docs/DEVELOPING.md:33]: It will be more clear if we rephrase the compatibility section to be something like:

Confluence Cloud or Confluence Server/Data Center versions 7 or later are compatible with Elastic connector frameworks.

File: docs/DEVELOPING.md
Comment (ppf2) [inline docs/DEVELOPING.md:57]: Do we support Confluence Data Center in addition to Confluence Server? If so, let's change all our "Confluence Server" references to "Confluence Server/Data Center".

File: docs/DEVELOPING.md
Comment (ppf2) [inline docs/DEVELOPING.md:31]: We are missing some fundamental information on what Confluence object types are supported by this connector.  For example, does it automatically fetch all pages, spaces, blogs, comments, templates, labels, attachments, categories, etc..?

File: docs/DEVELOPING.md
Comment (praveen-elastic) [reply docs/DEVELOPING.md]: No, the connector supports Confluence Cloud and Confluence Server only.

File: docs/DEVELOPING.md
Comment (leemthompo) [reply docs/DEVELOPING.md:31]: This info should go in **Documents and syncs** subheading :)

Comment (praveen-elastic) [reply docs/DEVELOPING.md:31]: Done :)

Comment (praveen-elastic) [reply docs/DEVELOPING.md:31]: Commit id: [7f6b6a5](https://github.com/elastic/connectors-python/pull/647/commits/7f6b6a511a3a472a49f4b362da7f15af88e57d96)

File: docs/DEVELOPING.md
Comment (leemthompo) [inline docs/DEVELOPING.md:97]:
```suggestion
The connector syncs the following Confluence object types: 
- Pages
- Spaces
- Blog Posts
- Attachments
```

**Review by leemthompo — DISMISSED** [review]
👍

**Review by ppf2 — APPROVED** [review]
Thanks for the changes!

**Comment by wangch079** [comment]
buildkite test this

**Pull request merged**
